8.0 refactor new api

lmignon opened the pull request:
The PR is a backport of odoo/odoo#5863.
The PR includes:
- Migration to new API
- Tests
- An orthogonal way to search for account_number whatever its formatting.  (sanitized_acc_number on res.partner.bank)
- A little refactoring of the import_file method to improve the extensibility (required for https://github.com/OCA/bank-statement-import/pull/14)

To ease the review, this PR doesn't include an interesting PR from @NL66278 on  odoo/odoo#5863 https://github.com/acsone/odoo/pull/2. 'Support import for different accounts and currencies.'

lmignon committed:
[FIX] bank_account created by the import must belong to the partner linked to the company of the provided journal
PR on master: odoo/odoo#5821

lmignon committed:
[IMP] Backport from master at 04daefd2d101d90daf5782173b95f31f3bd9e1b6

lmignon committed:
[IMP] Add sanitezed_acc_number on res.partenr.bank
The new field is computed by sanitzing the acc_number.

lmignon committed:
[IMP] Improve test coverage

lmignon committed:
[IMP] Sanitize acc_number to upper case
Use wildcard on both sides when opertor contains 'like'

lmignon committed:
[IMP] Let the res_partner_bank.post_write create the journal if it doesn't exist

lmignon committed:
[IMP] Specify the currency to use for the account.account and account.journal created by res_partner_bank.post_write

lmignon committed:
[FIX] backport from MP on upstream until 5f30c6ed09e92e53ad870d1fd54288ac43a4ca9e

Commit:
[FIX] Use unique accounts and improved bank search in base.

Commit:
[FIX] No index=True needed when specifying unique constraint.

lmignon committed:
[IMP] use new API

Conflicts:
	account_bank_statement_import/account_bank_statement_import.py

yvaucher commented:
@lmignon According to @qdp-odoo in odoo/odoo#5863 not much was kept from this PR on odoo.

Is this PR still valid?

sbidoul commented:
@yvaucher I understood that the concepts were kept on master even if the code ends up different.

See also my other comment about this https://github.com/OCA/bank-statement-import/issues/18#issuecomment-104199209.

I was the one who advocated to keep in sync with upstream for this part. I now think it's not realistic anymore. So I believe we should move forward while keeping an eye on master to make sure that at least the parsers remain compatible so migrating to v9 will be as painless as possible.

lmignon commented:
@yvaucher Yes it's. The description provides the improvements implemented in this PR and @sbidoul  explains in his latest comment why the PR diverges from the upstream.

yvaucher commented:
Ok then :+1:

pedrobaeza commented on account_bank_statement_import/__openerp__.py:
```diff
@@ -1,18 +1,16 @@
     'demo': [],
-    'description' : """Generic Wizard to Import Bank Statements.
-    
+    'description': """Generic Wizard to Import Bank Statements.
```
As we are going to diverge from master, please put this on a README.rst with the proper OCA format.

pedrobaeza commented on account_bank_statement_import_ofx/__openerp__.py:
```diff
@@ -25,9 +25,8 @@
 create periods for the year 2013.
```
Extract description

pedrobaeza commented on account_bank_statement_import_ofx/account_bank_statement_import_ofx.py:
```diff
@@ -14,41 +12,49 @@
 
-class account_bank_statement_import(osv.TransientModel):
+
+class account_bank_statement_import(models.TransientModel):
```
CamelCase

pedrobaeza commented on account_bank_statement_import_qif/__openerp__.py:
```diff
@@ -4,7 +4,7 @@
+    'category': 'Accounting & Finance',
     'version': '1.0',
     'author': 'OpenERP SA',
     'description': '''
```
Extract description

pedrobaeza commented on account_bank_statement_import_qif/account_bank_statement_import_qif.py:
```diff
@@ -6,41 +6,49 @@
-    _defaults = {
-        'hide_journal_field': _get_hide_journal_field,
-    }
+    journal_id = fields.Many2one(
```
Please put this field in the main module, because there are other parsers that can use it (spanish one, for example), and it makes no sense to duplicate it when you have all installed.

pedrobaeza commented:
Why changing the SVG file? I put one with the text converted to curves to avoid to need the font.

lmignon commented:
@pedrobaeza I'll take into account all your remarks ASAP. Once again Thank you for the review...

lmignon committed:
[IMP] Extract description to README
Add OCA as author,
PEP8

lmignon committed:
[ADD] i18n

lmignon committed:
[FIX] Keep icon retouched by @pedrobaeza

lmignon commented:
@pedrobaeza I've modified the code according to your remarks

NL66278 commented:
:+1:  Time to get this merged.

pedrobaeza commented:
There's still the field and view for putting journal_id in the main module. For us is totally needed to avoid the dependency on qif file.

lmignon commented:
@pedrobaeza IMO, this change can be done in an other PR once it'll be required to support a new parser.

pedrobaeza commented:
It is already required by our spanish bank statement file, that doesn't include bank information in it. It's already merged in https://github.com/OCA/l10n-spain/tree/8.0/l10n_es_account_bank_statement_import_n43.

Please let's make it in one step.

lmignon committed:
[IMP]Move the field journal_id in the main module.
By moving the field in the main module the others parsers can use it without duplicate it

lmignon commented:
@pedrobaeza done

pedrobaeza commented:
Thank you very much, you're great! :+1: 

Merging finally!

pedrobaeza commented:
When goes green..

lmignon commented:
Thank you @pedrobaeza :smirk:
